### PR TITLE
macOS: fixed getting decimal separator.

### PR DESCRIPTION
**Pull request by laptabrok**
Under macOS all C string-related functions work in **C locale** by default even if OS region use other locale (e.g. in Russia). This leads to runtime checks failure when processing localised strings w/o explicitly setting `wxLocale`. For example, generic `wxSpinCtrlDouble` uses `wxFloatingPointValidator` for user input checking where check for locale-dependent _decimal separator_ occurs. W/o this fix `wxNumberFormatter::GetDecimalSeparator()` could return '**,**' but `wxIntegerValidatorBase::FromString` would fail with that string-to-double conversion.

PS
I'd like to notice that on macOS, w/o explicitly setting wxLocale for user's current region, `wxString::FromDouble` will return string in C locale and `wxString::ToDouble` _will fail_ if we pass localised string like "120,034" which may be _correct_ for user's locale. I wrote macOS specific functions for conversion in that cases (not included in this PR) but I don't know is it a good practise to wxString locale-specific functions etc w/o wxLocale setup. I did not found any comments on this in the documentation.

### CODE REVIEW

**Comment by laptabrok** [comment]
We can also return current decimal separator used by standard C functions here instead of hardcoded decimal point. It may be useful in case of manual change of LC_NUMERIC.

**Comment by vadz** [comment]
Sorry, I don't understand why does this need to be done only for the decimal point. Surely the issue is exactly the same for any other locale-specific items?

It seems the real problem is that `CFLocaleCopyCurrent()` returns something different from the locale used by the standard library and I think we need to fix it there, i.e. somehow get `CFLocale` for "C" locale. @csomor Do you already know a way to do it by chance?

**Comment by vadz** [comment]
I'd like to apply this instead of this PR:
```diff
diff --git a/src/common/intl.cpp b/src/common/intl.cpp
index 0b0d8798f4..38837a55e2 100644
--- a/src/common/intl.cpp
+++ b/src/common/intl.cpp
@@ -1759,7 +1759,7 @@ wxString wxLocale::GetInfo(wxLocaleInfo index, wxLocaleCategory WXUNUSED(cat))
     }
     else // no current locale, use the default one
     {
-        userLocaleRefRaw = CFLocaleCopyCurrent();
+        userLocaleRefRaw = wxCFRetain(CFLocaleGetSystem());
     }

     wxCFRef<CFLocaleRef> userLocaleRef(userLocaleRefRaw);
```
Do you (or @csomor) see any problem with doing it?

**Comment by csomor** [comment]
@vadz Thanks, sorry, I forgot, I tried other things, in vain, as I also have similar problems for de_CH locale, as we use the point as decimal separator in Switzerland while Germany is using a comma. I'll try your approach

**Comment by csomor** [comment]
@vadz, now also the dateformats on my Swiss-German System at the default American formats, I don't think this is a good path. Where do we really rely on c functions for formatting locale aware data ?

**Comment by vadz** [comment]
Sorry, what exactly are you testing, i.e. do you use `wxLocale` or not? This just tries to avoid lying to the programs about the decimal separator (and other locale-specific items used) when they don't change the locale.

IMO the ideal behaviour should really be:

1. Use C (i.e. US) locale everywhere if `wxLocale` is not used at all.
2. Use user locale if `wxLocale(wxLANGUAGE_DEFAULT)` is used.

I think (2) should mostly work (except for the really bad problem described in [this ticket](https://trac.wxwidgets.org/ticket/19023) that I'd very much appreciate your feedback on too) and this tries to fix/improve things for (1).

**Comment by csomor** [comment]
I think we have to dive into these a litte bit more to make sure the cure is not worse than the problem

1) Right now wx under macOS is using the current CFLocale which is determined by the OS, so dateformats etc. are correct. Without setting a locale, not a c-level nor at wx level. 
Changing this now would suddenly change the behaviour of existing apps on macOS, eg Dates would be displayed differently ...

2) Setting a locale eg to de_CH is not working correctly at the c-level as I wrote, it only knows about de, the decimal and thousands separators are wrong. So there I'd like to leave LC_NUMERIC at C and perform the inverse of what FromDouble/FromCDouble etc are doing at wxString level.

So in summary IMHO we will never be able to depend on C-Level locale on macOS, we will have to use the CFLocale information and perform replacements like mentioned in 2)

**Comment by vadz** [comment]
We definitely need to discuss this, but I suggest doing it on wx-dev rather than in this PR because it's going to become unwieldy as the discussion goes on. I'm going to post my reply there, could you please take a look? TIA!

**Comment by vadz** [comment]
I believe this was made obsolete a long time ago by `wxUILocale` and the related changes.
